plugins/modules/accesspoint_workflow_manager.py

```diff
@@ -0,0 +1,521 @@
+            else:
+                responses["accesspoint"] = {
+                    "device_list": devices_list,
+                    "device_configs": ap_data,
+                    "final_input": final_ap_data,
+                    "msg": "Filtered AP Device data list for update AP config"}
+
+        self.result["response"].append(responses)
+        self.log("Current AP State "+ str(responses["accesspoint"]) , "INFO")
+        return final_ap_data
+
+    # below function not required for the current scenario but required for upcomming. 
+    def get_state(self):
+        """
+        This function used to get device details as json response from DNAC site based
+        on the config field given on the input yml file
+        Parameters:
+          - config:
+            -   mac_address: "90:e9:5e:03:f3:40" (Required Field)
+                led_brightness_level: 4 
+                led_status: "Enabled"
+                location: "LTTS"
+                ap_name: "NFW-AP1-9130AXE"
+        Returns:
+            {
+                "apEthernetMacAddress": "34:5d:a8:0e:20:b4",
+                "family": "Unified AP",
+                "hostname": "NFW-AP1-9130AXE",
+                "id": "37b05b0f-1b1e-496a-b101-8f277f0af8ff",
+                "lastUpdated": "2024-05-28 08:36:19",
+                "macAddress": "90:e9:5e:03:f3:40",
+                "managementIpAddress": "204.1.216.2",
+                "type": "Cisco Catalyst 9130AXE Unified Access Point",
+                "upTime": "08:22:37.080"
+            }
+        Example:
+            functions = Accesspoint(module)
+            device_data = functions.get_state()
+        """
+        self.log('Getting Network Device information', "INFO")
+        try:
+            all_mac_address = [eachdevice["mac_address"] for eachdevice in self.payload["config"]]
+            device_list = self.dnac.devices.get_device_list(macAddress=all_mac_address)
+            if self.payload["device_fields"] == "" or self.payload["device_fields"] == "all":
+                self.payload["device_list"] = device_list['response']
+                return self.payload["device_list"]
+
+            fields = [str(self.keymap[x]) for x in self.payload["device_fields"].split(",")]
+            if len(device_list) != 0:
+                df = pd.DataFrame.from_records(device_list['response'])
+                selected_data = df[fields]
+                self.payload["device_list"] = selected_data.to_dict('records')
+                return self.payload["device_list"]
+            else:
+                return None
+        except Exception as e:
+            self.log("Unable to get device info "+ str(e) , "ERROR")
+
+    def get_ap_configuration(self):
+        """
+        This function used to get AP device details as json response from DNAC site.
+        by giving MAC address as a input in the URL GET Method
+        Parameters:
```

Comment:
Used Args and define only the Args whatever it's in the API

Reply:
Added